sql/parser/encode.go

```diff
@@ -38,16 +38,37 @@ var (
 )
 
 func encodeSQLString(buf []byte, in []byte) []byte {
+	// Changed if escape sequences necessary. See:
+	// http://www.postgresql.org/docs/9.0/static/sql-syntax-lexical.html
+	var strMod byte
+	for _, ch := range in {
+		if encodedChar := encodeMap[ch]; encodedChar != dontEscape {
+			// If a character which we want to escape is found, update strMod.
+			strMod = 'e' // TODO(tschottdorf) should this be 'E'?
+			break
+		}
+	}
```

Comment:
These encoding routines were taken from vitess. They are also used in squalor where they were rewritten to append multiple non-encoded bytes at a time. Besides being more efficient, that structure might be easier to use here. Take a look at https://github.com/square/squalor/blob/master/encode.go#L100. The thought is to delay outputting the opening `'` until the first escaped character is seen.